k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -52,60 +57,75 @@ class EndpointsNamer(idPrefix: Path, mkApi: String => NsApi) extends Namer {
 
   private[this] object Ns {
     private[this] var caches: Map[String, NsCache] = Map.empty
+    // XXX once a namespace is watched, it is watched forever.
+    private[this] var _watches = Map.empty[String, Activity[Closable]]
+    def watches = _watches
+
+    /**
+      * Returns an Activity backed by a Future.  The resultant Activity is pending until the
+      * original future is satisfied.  When the Future is successful, the Activity becomes
+      * an Activity.Ok with a fixed value from the Future.  If the Future fails, the Activity
+      * becomes an Activity.Failed and the Future is retried with the given backoff schedule.
+      * Therefore, the legal state transitions are:
+      *
+      * Pending -> Ok
+      * Pending -> Failed
+      * Failed -> Failed
+      * Failed -> Ok
+      */
+    private[this] def retryToActivity[T](go: => Future[T]): Activity[T] = {
+      val state = Var(Activity.Pending: Activity.State[T])
```

Comment:
style: prefer Var[Activity.State[T]](Activity.Pending) (this is done throughout the codebase already)